cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -505,34 +505,31 @@ impl ProcessComposeLogReader {
                 };
             }
 
-            // Here, either sending to the receiver failed i.e. the receiver was dropped,
-            // or the child process died.
-            // If the child process ended, that's unexpected
-            // and we'll try to communicate that through the channel.
```

Comment:
I think this comment was a bit misleading as the error is returned via joining the handle, not the channel. Is my understanding correct?

Reply:
Your understanding is correct as far as I can tell.

Reply:
the comment was there to describe the two states that could break the loop, which is either the channel causing an error or the reader running out.
The misleading part about this is that `child process ended` was attributed to the fact that stdout was closed. That's probably largely correct still, as process-compose does not close stdout unless its dying, but it may do so before being completely gone which is why `try_wait` may not catch it in all cases.

Reply:
But we don't communicate that through the channel in either case, right?

Reply:
I think what @ysndr means is that we are communicating that via the channel (and it causing an error), but not by sending something over the channel.